src/core/state.go

```diff
@@ -988,6 +990,12 @@ func (state *BuildState) ForConfig(config ...string) *BuildState {
 	return s
 }
 
+func (state *BuildState) ForSubrepo(name string) *BuildState {
+	s := state.ForConfig()
+	s.CurrentSubrepo = name
```

Comment:
I wonder if we should do some check on whether `CurrentSubrepo == name` and return the existing one if so?

Reply:
Oh yeah, that's a good point.